collab-document/src/blocks/children.rs

```diff
@@ -0,0 +1,98 @@
+use collab::preclude::*;
+use serde::ser::SerializeMap;
+use serde::{Serialize, Serializer};
+
+pub struct ChildrenMap {
+  pub root: MapRefWrapper,
+}
+
+impl Serialize for ChildrenMap {
+  fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+  where
+    S: Serializer,
+  {
+    let txn = self.root.transact();
+    let mut map = serializer.serialize_map(Some(self.root.len(&txn) as usize))?;
+    for (key, _) in self.root.iter(&txn) {
+      let children = self.root.get_array_ref_with_txn(&txn, key).unwrap();
+      let value = serde_json::json!(children
+        .iter(&txn)
+        .map(|child| child.to_string(&txn))
+        .collect::<Vec<String>>());
+      map.serialize_entry(key, &value)?;
+    }
+    map.end()
+  }
+}
+
+impl ChildrenMap {
+  pub fn new(root: MapRefWrapper) -> Self {
+    Self { root }
+  }
+
+  pub fn to_json(&self) -> serde_json::Value {
+    serde_json::to_value(self).unwrap()
+  }
+
+  pub fn get_children_with_txn(
+    &self,
+    txn: &mut TransactionMut,
+    children_id: &str,
+  ) -> ArrayRefWrapper {
+    self
+      .root
+      .get_array_ref_with_txn(txn, children_id)
+      .unwrap_or_else(|| self.create_children_with_txn(txn, children_id.to_owned()))
+  }
+
+  pub fn create_children_with_txn(
+    &self,
+    txn: &mut TransactionMut,
+    children_id: String,
```

Comment:
If the passed-in parameter is must be String. then it's ok to declare the children_id is `String`. But here it should be `&str`.